lib/rspec/core/configuration.rb

```diff
@@ -2196,15 +2219,27 @@ def run_suite_hooks(hook_description, hooks)
         end
       end
 
```

Comment:
Gathering the intersection of files here is where I found it getting messy.  Specifically if there were ever to be more flags similar to these set, it might be worth extracting some of this functionality.